lib/errors/error-collector.js

```diff
@@ -89,75 +89,62 @@ class ErrorCollector {
   }
 
   /**
-   * Helper method for processing errors that are created with .noticeError()
+   * Gets the iterable property from the transaction based on the error type
    *
    * @param {Transaction} transaction the collected exception's transaction
-   * @param {number} collectedErrors the number of errors we've successfully .collect()-ed
-   * @param {number} expectedErrors the number of errors marked as expected in noticeError
-   * @returns {Array.<number>} the updated [collectedErrors, expectedErrors] numbers post-processing
+   * @param {string} errorType the type of error: "user", "transactionException", "transaction"
+   * @returns {Array} the iterable property from the transaction based on the error type
```

Comment:
It's a good idea to give a stronger indication of what the value is. In this case, an array of objects is returned, so we would use the `object[]` syntax to denote it:

```suggestion
   * @returns {object[]} the iterable property from the transaction based on the error type
```

https://jsdoc.app/tags-type explains the various syntaxes available for describing parameter types.